src/App.tsx

```diff
@@ -1,19 +1,141 @@
 import React from 'react';
-import './App.css';
+import './App.scss';
 
+import { TodoList } from './components/TodoList/TodoList';
+import TodoType from './types/TodoType';
+import todos from './api/todos';
 import users from './api/users';
 
-const App: React.FC = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+type State = {
+  allTodos: TodoType[];
+  userName: string;
+  todoItem: string;
+  error: string;
 };
 
-export default App;
+export default class App extends React.Component<{}, State> {
+  state:State = {
+    allTodos: [...todos],
+    userName: '',
+    todoItem: '',
+    error: '',
+  };
+
+  clearInputs = () => {
+    this.setState({
+      userName: '',
+      todoItem: '',
+      error: '',
+    });
+  };
+
+  addError = () => {
+    this.setState({
+      error: 'Please enter a valid value',
+    });
+  };
+
+  addTodo = ():any => {
+    const { userName, todoItem } = this.state;
+
+    if (
+      // if inputs are not empty
+      users.some(user => userName === user.name) && todoItem
+    ) {
+      this.setState(prevState => {
+        const currenTodoItem = prevState.todoItem;
+        const currentUserId = prevState.allTodos[prevState.allTodos.length - 1].id + 1;
+        const currentUserIndex = users.findIndex(user => prevState.userName === user.name);
+        const newTodo = {
+          userId: users[currentUserIndex].id,
+          id: currentUserId,
+          title: currenTodoItem,
+        };
+
+        return {
+          allTodos: [...prevState.allTodos, newTodo],
+        };
+      });
+      this.clearInputs();
+    } else {
+      this.addError();
+    }
+  };
+
+  render() {
+    const {
+      allTodos,
+      todoItem,
+      userName,
+      error,
+    } = this.state;
+
+    return (
+      <div className="app">
+        <h1 className="app__main-title">Todo list</h1>
+        <form
+          onSubmit={event => {
+            event.preventDefault();
+            this.addTodo();
+          }}
```

Comment:
add event.preventDefault inside the addTodo function

```suggestion
          onSubmit={this.addTodo}
```